optuna/testing/storages.py

```diff
@@ -98,7 +99,18 @@ def __enter__(
             self.thread = threading.Thread(target=self.server.start)
             self.thread.start()
 
-            return GrpcStorageProxy(host="localhost", port=port)
+            proxy = GrpcStorageProxy(host="localhost", port=port)
+
+            # Wait until the server is ready.
+            while True:
+                try:
+                    proxy.get_all_studies()
+                    break
+                except grpc.RpcError:
+                    time.sleep(1)
+                    continue
+
+            return proxy
```

Comment:
What about this?

> @c-bata 

Do you have any concern with this?

```suggestion
                    return proxy
                except grpc.RpcError:
                    time.sleep(1)
                    continue
```

Reply:
Either approach looks fine to me.